### ISSUE TITLE
Interfacing with reMarkable2 and trash items

### ISSUE THREAD

**Issue by ohad** [splitbrain/ReMarkableAPI#19]
The command line tool crashed with:
```
☠ Unknown ID trash. Inconsitent meta data
```
I think it's because I have deleted a file on the reMarkable2. It's a bit difficult to test, sorry about that.

The `--loglevel debug` shows:
```
-> GET https://document-storage-production-dot-remarkable-production.appspot.com/document-storage/json/2/docs
<- Status 200 5705 bytes
[
 ...
    {
        "ID": "c2a07c8d-2fa1-47ec-9522-b178f88f443e",
        "Version": 2,
        "Message": "",
        "Success": true,
        "BlobURLGet": "",
        "BlobURLGetExpires": "0001-01-01T00:00:00Z",
        "ModifiedClient": "2020-12-25T11:21:56.481481Z",
        "Type": "DocumentType",
        "VissibleName": "redacted.pdf",
        "CurrentPage": 0,
        "Bookmarked": false,
        "Parent": "trash"
    },
...
Exception caught in /home/ohad/dev/ReMarkableAPI/src/RemarkableFS.php:165
#0 /home/ohad/dev/ReMarkableAPI/src/RemarkableFS.php(176): splitbrain\RemarkableAPI\RemarkableFS->calcPath('trash')
#1 /home/ohad/dev/ReMarkableAPI/src/RemarkableFS.php(50): splitbrain\RemarkableAPI\RemarkableFS->calcPath('c2a07c8d-2fa1-4...')
#2 /home/ohad/dev/ReMarkableAPI/src/RemarkableFS.php(28): splitbrain\RemarkableAPI\RemarkableFS->refreshIndex()
#3 /home/ohad/dev/ReMarkableAPI/remarkable.php(117): splitbrain\RemarkableAPI\RemarkableFS->__construct(Object(splitbrain\RemarkableAPI\RemarkableAPI))
#4 /home/ohad/dev/ReMarkableAPI/remarkable.php(73): Remarkable->cmdList()
#5 /home/ohad/dev/ReMarkableAPI/vendor/splitbrain/php-cli/src/CLI.php(131): Remarkable->main(Object(splitbrain\phpcli\Options))
#6 /home/ohad/dev/ReMarkableAPI/remarkable.php(225): splitbrain\phpcli\CLI->run()
#7 {main}
```

**Comment by ohad** [comment]
Potentially addressed by PR #18 .
I'm not sure simply returning an empty path for the 'trash' ID is safe, though, since we end up with bogus paths

Perhaps something along these lines is safer?
```php
    protected function calcPath($id)
    {
        if (!isset($this->index[$id])) throw new \Exception("Unknown ID $id. Inconsitent meta data");
        $item =& $this->index[$id];

        // path already set?
        if (isset($item['Path'])) return $item['Path'];

        if ($item['Parent'] === '') {
            // top level item
            $item['Path'] = '/' . $item['VissibleName'];
        } else {if ($item['Parent'] === 'trash') {                 // <-------- Change starts here
           // trash item
           $item['Path'] = 'trash/' . $item['VissibleName'];
        } else {
            // recursion
            $item['Path'] = $this->calcPath($item['Parent']) . '/' . $item['VissibleName'];
        }}

        return $item['Path'];
    }
```
The above, while works for me, for now, is probably not a complete solution, just a workaround, since I've not made any attempts at understanding how trash works.

**Comment by splitbrain** [comment]
fixed with de5da2ac9316b4985e29f315efc670c89b2edeb6
